src/extensions/mountainview/MountainView/ViewContainer.tsx

```diff
@@ -1,46 +1,117 @@
+        const dropArea = (y < tabBarHeight + H0) ? 'north' : 'south'
+        const viewId = e.dataTransfer.getData('viewId')
+        if (viewId) {
+            const view = views.filter(v => v.viewId === viewId)[0]
+            if (view) {
+                if (view.area !== dropArea) {
+                    onSetViewArea(view, dropArea)
+                }
+            }
+        }
+    }, [views, H0, onSetViewArea])
+
+
     if (!Array.isArray(children)) {
-        throw Error('Unepected children in ViewContainer')
+        throw Error('Unexpected children in ViewContainer')
     }
     const children2 = children as React.ReactElement[]
-    const W = width || size.width || 300
+    const W = width || 300
     return (
-        <div style={{width: W}}>
-            <ViewContainerTabBar
-                views={views}
-                currentView={currentView}
-                onCurrentViewChanged={setCurrentView}
-                onViewClosed={onViewClosed}
-            />
+        <div
+            style={{width: W, height: height}} onClick={handleClick}
+            onDragOver={handleDragOver}
+            onDrop={handleDragDrop}
+        >
+            <div key="north-tab-bar" style={{position: 'absolute', ...areas['north'].tabBarStyle, border: 1, borderStyle: 'solid', borderColor: (activeArea === 'north') ? 'black' : 'white'}}>
+                <ViewContainerTabBar
+                    views={views.filter(v => v.area === 'north')}
+                    currentView={currentNorthView}
+                    onCurrentViewChanged={setCurrentNorthView}
+                    onViewClosed={onViewClosed}
+                />
+            </div>
+            <div key="south-tab-bar" style={{position: 'absolute', ...areas['south'].tabBarStyle, border: 1, borderStyle: 'solid',  borderColor: (activeArea === 'south') ? 'black' : 'white'}}>
+                <ViewContainerTabBar
+                    views={views.filter(v => v.area === 'south')}
+                    currentView={currentSouthView}
+                    onCurrentViewChanged={setCurrentSouthView}
+                    onViewClosed={onViewClosed}
+                />
+            </div>
             {
                 children2.map(c => {
                     const childView = c.props.view as any as View
-                    const visible = (childView === currentView)
+                    const visible = ((childView.area === 'north') && (childView === currentNorthView)) || ((childView.area === 'south') && (childView === currentSouthView))
+                    const area = areas[childView.area || 'north']
                     return (
-                        <div key={childView.viewId} style={{visibility: visible ? 'visible' : 'hidden', position: 'absolute', left: 0, width: W, top: tabBarHeight}}>
+                        <div key={childView.viewId} style={{visibility: visible ? 'visible' : 'hidden', overflowY: 'auto', overflowX: 'hidden', position: 'absolute', ...area.divStyle}}>
```

Comment:
While it would seem natural for the north and south tab widgets to be DOM elements containing their views, this would break the state of the views when the view is moved between north and south. We keep react happy by having all views as children of the main container. We just position them differently depending on their area property.